### PR TITLE
chore: run website tests against netlify deployment

### PR DESCRIPTION
**Pull request by JamesHenry**
Rather than running an extra dev-server on localhost within the Github agent, which has proven to be flaky, this runs the website tests against the actual Netlify deployment which we are doing anyway as part of our workflow.

In my initial implementation I had a very convoluted way of achieving it via Netlify build hooks and a webhook handler which triggered an additional workflow.

It has proven to be much simpler by leveraging the Netlify API instead (which was already done by an existing Github Action from the community)

### CODE REVIEW

**Comment by JamesHenry** [comment]
Oh Netlify still doesn't dispatch `deployment_status` events even many years after it has been available and has been requested. That's very surprising.

We can potentially leverage an HTTP trigger for this though

File: packages/website/tests/index.spec.ts [hunk 1]
```diff
@@ -15,8 +15,6 @@ test.describe('Website', () => {
       }
     });
     await page.goto('/');
-    expect(errorMessages).toStrictEqual([
-      "[error] Warning: ReactDOM.render is no longer supported in React 18. Use createRoot instead. Until you switch to the new API, your app will behave as if it's running React 17. Learn more: https://reactjs.org/link/switch-to-createroot",
```
Comment (JamesHenry) [inline packages/website/tests/index.spec.ts:19]: This error message was only applicable to the local dev-server and is not shown to our users on production builds. A nice extra benefit of this approach is that our testing is more representative of our end UX

**Comment by JamesHenry** [comment]
The remaining website test failure is consistent with the same failure on the latest main run, so it is not related to the changes

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -179,26 +179,38 @@ jobs:
       contents: read # to fetch code (actions/checkout)
 
     name: Website tests
+    # We technically do not need to wait for build within the pipeline any more because the build we care about is happening within Netlify, however,
+    # it is highly likely that if the CI one fails, the Netlify one will as well, so in order to not waste unncessary Github Actions minutes/resources,
+    # we do still keep this requirement here.
```
Comment (JoshuaKGoldberg) [inline .github/workflows/ci.yml:184]: _chuckles in #5573_

**Review by JoshuaKGoldberg — APPROVED** [review]
⚡ yess!!!

**Pull request merged**